v2/offset_reader.go

```diff
@@ -0,0 +1,59 @@
+package car
+
+import "io"
+
+// OffsetReader implements Read, and ReadAt on a section
+// of an underlying io.ReaderAt.
+// The main difference between io.SectionReader and OffsetReader is that
+// NewOffsetReader accepts zero as n, the number of bytes to read, with
+// the trade-off that it does not implement Seek.
+// When n is set to zero, it will delegate io.EOF errors to the underlying
+// io.ReaderAt.
+// This is useful when reading a section at the end of a io.ReaderAt without
+// having to know the total number readable of bytes.
+type OffsetReader struct {
+	r       io.ReaderAt
+	base    int64
+	off     int64
+	limit   int64
+	limited bool
+}
+
+// NewOffsetReader returns an OffsetReader that reads from r
+// starting at offset off and stops with io.EOF after n bytes.
+// If n is set to 0 then it will carry on reading until r reaches io.EOF.
+func NewOffsetReader(r io.ReaderAt, off int64, n int64) *OffsetReader {
```

Comment:
not sure I understand the point of `n int64`, since it's always going to be 0. if it were > 0, then one could just use io.SectionReader, setting the limit to off+n.

so I think you should remove that parameter, and remove the `limited` boolean as it's always true.